D10 - Default Paragraphs

CharlesJ3 opened the pull request:
Default Paragraphs according to link below and their dependencies.

https://denveru.sharepoint.com/:x:/r/sites/DrupalDevelopment/_layouts/15/Doc.aspx?sourcedoc=%7B0685A75C-F060-46D3-B729-3773C4946FC2%7D&file=paragraph_types.xlsx&action=default&mobileredirect=true

kreynen-du commented on web/profiles/custom/ducore/config/install/core.entity_form_display.paragraph.cards.default.yml:
```diff
@@ -0,0 +1,60 @@
+uuid: c503972f-a527-4949-9f06-cff44c29fd3f
```
One of the differences you'll notice between the yml files exported with config_profile like https://github.com/University-of-Denver/drupal-composer-managed/blob/master/web/profiles/custom/ducore/config/install/config_ignore.settings.yml#L1 and the site config export is the config_profile exports don't have uuids. [Trying to re-use uuids for multiple sites is a recipe for a bad time in the future](https://www.drupal.org/docs/administering-a-drupal-site/configuration-management/managing-your-sites-configuration#s-drupal-site-uuid).

CharlesJ3 replied:
Thanks for that information, some shared configuration in the D9 does indeed have the same unique uuid so this is helpful to know for this and future projects! 

I removed ALL UUIDs I could find from the .yml files I originally uploaded.

kreynen-du commented on composer.json:
```diff
@@ -215,6 +215,51 @@
             }
+        },
+        {
+            "_webform": true,
```
Why is webform true set here?

CharlesJ3 replied:
I removed the webform setting in all three of the libraries I added in composer.json, they are on all of the other dependencies so I assumed incorrectly they would be needed here.

kreynen-du commented on web/profiles/custom/ducore/modules/custom/du_site/du_site.info.yml:
```diff
@@ -0,0 +1,7 @@
+package: "DU Custom"
+core_version_requirement: ^8 || ^9 || ^10
+dependencies:
+  - du_widen
```
We are not sure we are going to use Widen on all sites. It should not be a requirement until that decision is made.

CharlesJ3 replied:
Understood, I ended up removing the entire du_site module because it was being used for one small piece of the "Block" paragraph, which we can just recreate in this new du_paragraphs module if needed.

kreynen-du commented on web/profiles/custom/ducore/modules/custom/du_site/du_site.install:
```diff
@@ -0,0 +1,559 @@
+<?php
+
+/**
```
None of these functions run on new D10 installs and includes content types and codes that will not be part of DU Core profile all sites have in common.

kreynen-du reviewed (changes requested):
The goal of this was really to create a module similar to https://www.drupal.org/project/varbase_bootstrap_paragraphs that included the most commonly used DU paragraphs to keep that separate from the DU Core profile. While the custom module would be in the profile for now (like du_widen), it would only be a requirement of pl_drupal for now.  A dependency that is specific to the MVP paragraphs should be a dependency of that module... not the upstream or profile.

It is very likely we will shift away from a large number of the current paragraphs after we complete the D9->D10 migration.

CharlesJ3 commented:
Thanks for the detailed comments @kreynen-du I'll get to work on this and have an update for you when we meet tomorrow morning.

CharlesJ3 commented:
Thanks again @kreynen-du , some updates added for each comment. 

Overall I moved the configuration files to a new module (du_paragraphs) and removed the following paragraphs:

Block, Contact, Contact Single, and Top News Story. Block required du_site and the rest required Content Types (Article, Contact). 

This works locally when running a site profile install (drush site-install ducore). Let me know if you find any issues and I'll get them fixed ASAP.
